unit_tests/log_transfer/test_main_prior_list_serialize.cpp

```diff
@@ -0,0 +1,489 @@
+void
+test_env::log_append_undo (const DataType &data, const std::string &undo)
+{
+  log_append_undoredo (data, undo, std::string ());
+}
+
+template <typename DataType>
+void
+test_env::log_append_noundoredo (const DataType &data)
+{
+  log_append_undoredo (data, std::string (), std::string ());
+}
+
+void
+test_env::serialize_deserialize_list ()
+{
+  if (m_prior_lsa_info.prior_list_header == nullptr)
+    {
+      return;
+    }
+
+  // pull list
+  log_prior_node *list_headp = m_prior_lsa_info.prior_list_header;
+  m_prior_lsa_info.prior_list_header = nullptr;
+  m_prior_lsa_info.prior_list_tail = nullptr;
+
+  // serialize/deserialize
+  std::string serialized = prior_list_serialize (list_headp);
+  log_prior_node *deserialized_headp = prior_list_deserialize (serialized);
+
+  // compare lists
+  log_prior_node *list_nodep = list_headp;
+  log_prior_node *deserialized_nodep = deserialized_headp;
+  while (list_nodep != nullptr)
+    {
+      REQUIRE (deserialized_nodep != nullptr);
+      require_equal (list_nodep, deserialized_nodep);
+      list_nodep = list_nodep->next;
+      deserialized_nodep = deserialized_nodep->next;
+    }
+
+  free_list (list_headp);
+  free_list (deserialized_headp);
+}
+
+void
+test_env::free_list (log_prior_node *listp)
+{
+  log_prior_node *save_next = nullptr;
+  for (log_prior_node *nodep = listp; nodep != nullptr; nodep = save_next)
+    {
+      save_next = nodep->next;
+
+      if (nodep->data_header != nullptr)
+	{
+	  free (nodep->data_header);
+	}
+      if (nodep->udata != nullptr)
+	{
+	  free (nodep->udata);
+	}
+      if (nodep->rdata != nullptr)
+	{
+	  free (nodep->rdata);
+	}
+      free (nodep);
+    }
+}
+
+void
+test_env::require_equal (log_prior_node *left, log_prior_node *rite)
+{
+  REQUIRE (left->log_header.back_lsa == rite->log_header.back_lsa);
+  REQUIRE (left->log_header.forw_lsa == rite->log_header.forw_lsa);
+  REQUIRE (left->log_header.prev_tranlsa == rite->log_header.prev_tranlsa);
+  REQUIRE (left->log_header.trid == rite->log_header.trid);
+
+  REQUIRE (left->start_lsa == rite->start_lsa);
+  REQUIRE (left->tde_encrypted == rite->tde_encrypted);
+
+  REQUIRE (left->data_header_length == rite->data_header_length);
+  require_mem_equal (left->data_header, rite->data_header, left->data_header_length);
+
+  REQUIRE (left->ulength == rite->ulength);
+  require_mem_equal (left->udata, rite->udata, left->ulength);
+
+  REQUIRE (left->rlength == rite->rlength);
+  require_mem_equal (left->rdata, rite->rdata, left->rlength);
+}
+
+void
+test_env::require_mem_equal (const char *memleft, const char *memrite, size_t memsz)
+{
+  if (memsz == 0)
+    {
+      REQUIRE (memleft == nullptr);
+      REQUIRE (memrite == nullptr);
+    }
+  const char *leftp = memleft;
+  const char *ritep = memrite;
+  for ( ; leftp < memleft + memsz; ++leftp, ++ritep)
+    {
+      REQUIRE (*leftp == *ritep);
+    }
+}
+
+//
+// Definitions of CUBRID stuff that is not used, but is needed for compile
```

Comment:
technically, these are needed by the linker :)

Reply:
right, comment changed